proxy/round_tripper/proxy_round_tripper.go

```diff
@@ -391,6 +424,21 @@ func (rt *roundTripper) selectEndpoint(iter route.EndpointIterator, request *htt
 	return endpoint, nil
 }
 
+func checkResponseHeaders(maxCount int, headers http.Header) error {
+	if maxCount > 0 {
+		hdrCount := 0
+		for _, vv := range headers {
+			hdrCount += len(vv)
```

Comment:
a small comment here would be nice: we count all instances of repeated headers, and all comma separated lists. According to the RFC they're equivalent.

This should be equivalent to the comments on the request size, which explain nicely what happens and what we limit.

Reply:
To make sure we are on the same page, this:

```
Foo: bar
Foo: bar, baz
```

will result in:

```go
map[string][]string{
	"Foo": []string{"bar", "bar, baz"},
}
```

To my understanding this is in conflict with the RFC but there's probably a good reason for that? At any rate: this allows us to deduce the number of header keys which for us is good, otherwise we would have no way of knowing how often a header key occurred.

In any case: here we are just interested in the number of keys which is equal to the number of values in the entire map. I'm adding a comment similar to the one present in the MaxRequestSize middleware.

Reply:
you're right. I wrote the thing about the RFC, then read the comment on the other code that explains what's happening and that it's counting the number of values, not the number of "header lines".

Either way, as long as it's consistent it's good.

adding the comment is what I wanted. Thanks!